test/python/common.py

```diff
@@ -174,4 +174,31 @@ def _is_ci_fork_pull_request():
     return False
 
 
+def compare_dicts(dict1, dict2, threshold, default=0):
+    """
+    Compare two dictionaries with numeric values.
```

Comment:
Can you also expand the docstring adding an explanation on what the behaviour is when `dict1` and `dict2` have a different set of keys, for clarity?